core/domain/user_jobs_one_off.py

```diff
@@ -18,12 +18,50 @@
 
 from core import jobs
 from core.domain import subscription_services
+from core.domain import user_services
 from core.platform import models
 (exp_models, collection_models, feedback_models, user_models) = (
     models.Registry.import_models([
         models.NAMES.exploration, models.NAMES.collection,
         models.NAMES.feedback, models.NAMES.user]))
 
+class UserContributionsOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for creating and populating UserContributionsModels for 
+    all registered users that have contributed.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationSnapshotMetadataModel]
+
+    @staticmethod
+    def map(item):
+
+        split_id = item.id.split("-")
+        yield (item.committer_id, {
+            'version_number': split_id[1],
+            'exploration_id': split_id[0]
+        })
+
+    @staticmethod
+    def reduce(key, version_and_exp_ids):
+
+        created_explorations = []
+        edited_explorations = []
+
+        edits = [ast.literal_eval(v) for v in version_and_exp_ids]
+
+        for edit in edits:
+            edited_explorations.append(edit['exploration_id'])
+            if edit['version_number'] == '1':
+                created_explorations.append(edit['exploration_id'])
+
+
+        if user_services.get_user_contributions(key, strict=False) is not None:
+            user_services.update_user_contributions(key, list(set(created_explorations)),
```

Comment:
Break line after '(' (so that the next line is indented by 4 more and starts with 'key')